packages/flutter_tools/test/src/fake_process_manager.dart

```diff
@@ -382,4 +388,35 @@ class _SequenceProcessManager extends FakeProcessManager {
 
   @override
   bool get hasRemainingExpectations => _commands.isNotEmpty;
+
+  @override
+  List<FakeCommand> get _remainingExpectations => _commands;
+}
+
+/// Matcher that successfully matches against a [FileSystemEntity] that
+/// exists ([FileSystemEntity.existsSync] returns true).
+const Matcher hasNoRemainingExpectations = _HasNoRemainingExpectations();
+
+class _HasNoRemainingExpectations extends Matcher {
+  const _HasNoRemainingExpectations();
+
+  @override
+  bool matches(dynamic item, Map<dynamic, dynamic> matchState) =>
+      item is FakeProcessManager && !item.hasRemainingExpectations;
+
+  @override
+  Description describe(Description description) =>
+      description.add('a fake process manager with no remaining expectations');
+
+  @override
+  Description describeMismatch(
+      dynamic item,
+      Description description,
+      Map<dynamic, dynamic> matchState,
+      bool verbose,
+      ) {
+    final FakeProcessManager fakeProcessManager = item as FakeProcessManager;
+    return description.add(
```

Comment:
`Iterable.join` will call toString on each element first, so you can simplify this to:

`fakeProcessManager._remainingExpectations.join('\n')`

Reply:
I need `command.command`.  Your suggestion gives:
```
   Which: has remaining expectations:
          Instance of 'FakeCommand'
```
But we want:
```
   Which: has remaining expectations:
          [git, describe]
```
You're right though I don't need the `toString`.